homeassistant/components/overkiz/climate_entities/somfy_thermostat.py

```diff
@@ -0,0 +1,167 @@
+"""Support for TaHoma Smart Thermostat."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    PRESET_AWAY,
+    PRESET_HOME,
+    PRESET_NONE,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+PRESET_FREEZE = "freeze"
+PRESET_NIGHT = "night"
+
+STATE_DEROGATION_ACTIVE = "active"
+STATE_DEROGATION_INACTIVE = "inactive"
+
+
+OVERKIZ_TO_HVAC_MODES: dict[str, HVACMode] = {
+    STATE_DEROGATION_ACTIVE: HVACMode.HEAT,
+    STATE_DEROGATION_INACTIVE: HVACMode.AUTO,
+}
+HVAC_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODES.items()}
+
+OVERKIZ_TO_PRESET_MODES: dict[OverkizCommandParam, str] = {
+    OverkizCommandParam.AT_HOME_MODE: PRESET_HOME,
+    OverkizCommandParam.AWAY_MODE: PRESET_AWAY,
+    OverkizCommandParam.FREEZE_MODE: PRESET_FREEZE,
+    OverkizCommandParam.MANUAL_MODE: PRESET_NONE,
+    OverkizCommandParam.SLEEPING_MODE: PRESET_NIGHT,
+    OverkizCommandParam.SUDDEN_DROP_MODE: PRESET_NONE,
+}
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+TARGET_TEMP_TO_OVERKIZ = {
+    PRESET_HOME: OverkizState.SOMFY_THERMOSTAT_AT_HOME_TARGET_TEMPERATURE,
+    PRESET_AWAY: OverkizState.SOMFY_THERMOSTAT_AWAY_MODE_TARGET_TEMPERATURE,
+    PRESET_FREEZE: OverkizState.SOMFY_THERMOSTAT_FREEZE_MODE_TARGET_TEMPERATURE,
+    PRESET_NIGHT: OverkizState.SOMFY_THERMOSTAT_SLEEPING_MODE_TARGET_TEMPERATURE,
+}
+
+
+class SomfyThermostat(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Smart Thermostat."""
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+    _attr_hvac_modes = [*HVAC_MODES_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    # Both min and max temp values have been retrieved from the Somfy Application.
+    _attr_min_temp = 15.0
+    _attr_max_temp = 26.0
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(2)
```

Comment:
Where does the `2` come from?

Reply:
Overkiz returns several json object that’s look like 

```JSON
{
"creationTime": 1576009624000,
  "lastUpdateTime": 1576009624000,
  "label": "Humidité Salon",
  "deviceURL": "somfythermostat://SOMFY_THERMOSTAT-2189/184210979221#3",
  "shortcut": false,
  "controllableName": "somfythermostat:SomfyThermostatHumiditySensor"
  "...": "..."
}
```

Normally, each object is a device, but for some like the thermostat, Overkiz split the device in various object. Objects linked to the same device have the same device url except the digit after the `#`. In my example, it’s 3. And for the temperature, it’s 3:

```
{
"creationTime": 1576009624000,
  "lastUpdateTime": 1576009624000,
  "label": "Température Salon",
  "deviceURL": "somfythermostat://SOMFY_THERMOSTAT-2189/184210979221#2",
  "shortcut": false,
  "controllableName": "somfythermostat:SomfyThermostatTemperatureSensor"
  "...": "..."
}
```

Reply:
OK, I see.

I'd suggest to use a constant instead of the magic number 2 directly when fetching the correct device/object then.
Also, is some validation needed to make sure we get the correct object at the requested index?